lib/pacemaker/pcmk_sched_instances.c

```diff
@@ -778,6 +778,77 @@ unassign_if_mandatory(pe_action_t *first, pe_action_t *then,
     return false;
 }
 
+/*!
+ * \internal
+ * \brief Find first matching action for a clone instance or bundle container
+ *
+ * \param[in] action       Action in an interleaved ordering
+ * \param[in] instance     Clone instance or bundle container being interleaved
+ * \param[in] action_name  Action to look for
+ * \param[in] node         If not NULL, require action to be on this node
+ * \param[in] for_first    If true, \p instance is the 'first' resource in the
+ *                         ordering, otherwise it is the 'then' resource
+ *
+ * \return First action for \p instance (or its containerized resource if a
+ *         bundle container) that matches \p action_name and \p node if any,
+ *         otherwise NULL
+ */
+static pe_action_t *
+find_instance_action(const pe_action_t *action, const pe_resource_t *instance,
+                     const char *action_name, const pe_node_t *node,
+                     bool for_first)
+{
+    const pe_resource_t *rsc = NULL;
+    pe_action_t *matching_action = NULL;
+
+    /* If instance is a bundle container, sometimes we should interleave the
+     * action for the container itself, and sometimes for the containerized
+     * resource.
+     *
+     * For example, given "start bundle A then bundle B", B likely requires the
+     * service inside A's container to be active, rather than just the
+     * container, so we should interleave the action for A's containerized
+     * resource. On the other hand, it's possible B's container itself requires
+     * something from A, so we should interleave the action for B's container.
+     *
+     * Essentially, for 'first', we should use the containerized resource for
+     * everything except stop, and for 'then', we should use the container for
+     * everything except promote and demote.
+     */
+    if ((for_first && !pcmk__str_any_of(action->task, CRMD_ACTION_STOP,
+                                        CRMD_ACTION_STOPPED, NULL))
+
+        || (!for_first && pcmk__str_any_of(action->task, CRMD_ACTION_PROMOTE,
```

Comment:
We dropped the comment that clarified why we use the containerized resource for promote/demote. It's fairly clear if you think about it, but we may want to keep a quick note.

Reply:
true